src/v3/bandwidthRtc.ts

```diff
@@ -346,29 +378,88 @@ export class BandwidthRtc {
     });
   }
 
-  private setupPublishingPeerConnection(): RTCPeerConnection {
+  private async addHeartbeatDataChannel(peerConnection: RTCPeerConnection): Promise<RTCDataChannel> {
+    //Create a heartbeat data channel
+    let heartbeatDataChannel = peerConnection.createDataChannel('__heartbeat__', {
+      id: 0,
+      negotiated: true,
+      protocol: 'udp'
+    });
+    heartbeatDataChannel.onmessage = function(event) {
```

Comment:
Stylistically `function` doesn't tend to be used in this project for whatever reason.
`heartbeatDataChannel.onmessage = (event) => {`

Same below.